kornia/geometry/ransac.py

```diff
@@ -0,0 +1,224 @@
+        return model_best, inliers_best, best_model_score
+
+    def remove_bad_samples(self, kp1: torch.Tensor, kp2: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """"""
+        # ToDo: add (model-specific) verification of the samples,
+        # E.g. contraints on not to be a degenerate sample
+        return kp1, kp2
+
+    def remove_bad_models(self, models: torch.Tensor) -> torch.Tensor:
+        # ToDo: add more and better degenerate model rejection
+        # For now it is simple and hardcoded
+        main_diagonal = torch.diagonal(models,
+                                       dim1=1,
+                                       dim2=2)
+        mask = main_diagonal.abs().min(dim=1)[0] > 1e-6
+        return models[mask]
+
+    def polish_model(self,
+                     kp1: torch.Tensor,
+                     kp2: torch.Tensor,
+                     inliers: torch.Tensor) -> torch.Tensor:
+        # ToDo: Replace this with MAGSAC++ polisher
+        kp1_inl = kp1[inliers][None]
+        kp2_inl = kp2[inliers][None]
+        num_inl = kp1_inl.size(1)
+        model = self.polisher_solver(kp1_inl,
+                                     kp2_inl,
+                                     torch.ones(1,
+                                                num_inl,
+                                                dtype=kp1_inl.dtype,
+                                                device=kp1_inl.device))
+        return model
+
+    def forward(self,
+                kp1: torch.Tensor,
+                kp2: torch.Tensor,
+                weights: Optional[torch.Tensor] = None) -> Tuple[torch.Tensor, torch.Tensor]:
+        '''
+            Args:
+                kp1 (torch.Tensor): source image keypoints :math:`(N, 2)`.
+                kp2 (torch.Tensor): distance image keypoints :math:`(N, 2)`.
+                weights (torch.Tensor): optional correspondences weights. Not used now
+
+            Returns:
+                - estimated model, shape of :math:`(1, 3, 3)`.
+                - the inlier/outlier mask, shape of :math:`(1, N), where N is number of input correspondences`.'''
+        if not isinstance(kp1, torch.Tensor):
+            raise TypeError(f"Input kp1 is not torch.Tensor. Got {type(kp1)}")
+        if not isinstance(kp2, torch.Tensor):
+            raise TypeError(f"Input kp2 is not torch.Tensor. Got {type(kp2)}")
+        if not len(kp1.shape) == 2:
+            raise ValueError(f"Invalid kp1 shape, we expect Nx2 Got: {kp1.shape}")
+        if not len(kp2.shape) == 2:
+            raise ValueError(f"Invalid kp2 shape, we expect Nx2 Got: {kp2.shape}")
+        if not (kp1.shape[0] == kp2.shape[0]) or (kp1.shape[0] < self.minimal_sample_size):
+            raise ValueError(f"kp1 and kp2 should be \
+                             equal shape at at least [{self.minimal_sample_size}, 2], \
+                             got {kp1.shape}, {kp2.shape}")
+
+        best_score_total: float = 1.0
+        num_tc: int = len(kp1)
```

Comment:
what `tc` stands for ? find a more explicit name for that variable

Reply:
tentative correspondences